src/StoragelessSession/Session/Data.php

```diff
@@ -33,14 +33,50 @@ class Data implements \JsonSerializable
     private $metadata;
 
     /**
+     * @var bool
+     */
+    private $dataWasChanged = false;
```

Comment:
Why not just "changed"?

Translating to code:

```php
// I think that is a bit weird to ask:

$data->isDataChanged();

// Externally we don't care how the object handles itself
// we just want to know if it was changed, so this would
// be more "natural" IMHO:

$data->isChanged();
```